frontend/src/app/raw-preprocess-table/raw-preprocess-table.component.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  Component,
+  OnInit,
+  Output,
+  EventEmitter,
+  ViewChild,
+} from '@angular/core';
+import { HttpService, ResponseFileInfo } from '../services/http.service';
+import { Subscription, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { HttpErrorResponse } from '@angular/common/http';
+import { MatTableDataSource } from '@angular/material/table';
+import { MatSort } from '@angular/material/sort';
+import { MatPaginator } from '@angular/material/paginator';
+
+export interface PreprocessEntry {
+  name: string;
+  status: string;
+}
+
+export interface FileInfo {
+  name: string;
+  preprocessed: boolean;
+  loading: boolean;
+}
+
+@Component({
+  selector: 'raw-preprocess-table',
+  templateUrl: './raw-preprocess-table.component.html',
+  styleUrls: ['./raw-preprocess-table.component.scss'],
+})
+export class RawPreprocessTableComponent implements OnInit {
+  @Output() message = new EventEmitter<string>();
+  subscription: Subscription;
```

Comment:
Remember to unsubscribe observables at ngOnDestory. Please add corresponding logic to prevent memory leak